src/js/client/templateFormControl.js

```diff
@@ -0,0 +1,215 @@
+/*
+
+  A component to standardize handling of "simple" form data submitted via AJAX.  This component:
+
+  1.  Performs an initial render of the component using the template specified in `options.templates.initial`
+  2.  Binds a form submission that sends the `model` data using the options specified in `options.ajax`
+  3.  If the AJAX request returns an error or the result contains a "falsy" `ok` variable:
+      a)  Any previous "success" messages are cleared.
+      b)  the error is transformed using the rules found in `options.rules.error`
+      c)  the results are passed along to the `error` subcomponent for display.
+  4.  If the AJAX request is successful:
+      a)  Any previous "error" messages are cleared.
+      b)  The results are transformed using the rules found in `options.rules.success` and passed to the `success` subcomponent for display.
+      c)  The results are transformed using the rules found in `options.rules.model`, and applied to the model using the change applier.
+
+ */
+/* global fluid, jQuery */
+(function ($) {
+    "use strict";
+    var gpii = fluid.registerNamespace("gpii");
+    fluid.registerNamespace("gpii.templates.hb.client.templateFormControl");
+
+
+    // TODO: Replace this with JSON Schema validation: https://issues.gpii.net/browse/GPII-1176
+    gpii.templates.hb.client.templateFormControl.checkRequirements = function (that) {
+        var errors = [];
+
+        if (!that.options.templates || !that.options.templates.initial || !that.options.templates.error || !that.options.templates.success) {
+            errors.push("You have not configured any templates for use with this component.");
```

Comment:
No worries, @amb26, I saw it.  I plan to extract the JSON Schema validation components from the UL and PTD and standardize on using JSON Schema for client and server side validation.  I will ask Steve to create a repo to hold that and track the pull request.